packages/cli/src/metrics/index.ts

```diff
@@ -4,68 +4,170 @@ import { N8N_VERSION } from '@/constants';
 import * as ResponseHelper from '@/ResponseHelper';
 import type express from 'express';
 import promBundle from 'express-prom-bundle';
-import promClient from 'prom-client';
+import type { Counter } from 'prom-client';
 import semverParse from 'semver/functions/parse';
+import { Service } from 'typedi';
+import EventEmitter from 'events';
+import { CacheService } from '@/services/cache.service';
+import { MetricsCounterEvents } from './constants';
+import type { EventMessageTypes } from '@/eventbus/EventMessageClasses';
+import { LoggerProxy } from 'n8n-workflow';
+import { getLabelsForEvent } from '@/eventbus/MessageEventBusDestination/Helpers.ee';
+import { eventBus } from '../eventbus';
 
-export { promClient };
+@Service()
+export class MetricsService extends EventEmitter {
+	constructor(private readonly cacheService: CacheService) {
+		super();
+	}
+
+	// eslint-disable-next-line @typescript-eslint/consistent-type-imports
+	promClient: typeof import('prom-client') | undefined;
+
+	counters: Record<string, Counter<string> | null> = {};
+
+	async configureMetrics(app: express.Application) {
+		if (!config.getEnv('endpoints.metrics.enable')) {
+			return;
+		}
 
-export function configureMetrics(app: express.Application) {
-	if (!config.getEnv('endpoints.metrics.enable')) {
-		return;
+		this.promClient = await import('prom-client');
```

Comment:
sorry for the confusion. I meant that we should make the import of `MetricsService` in `Server.ts` dynamic. making this import dynamic isn't necessary.